Zend/zend_builtin_functions.c

```diff
@@ -1273,9 +1270,19 @@ ZEND_FUNCTION(get_object_vars)
 						const char *prop_name, *class_name;
 						size_t prop_len;
 						zend_unmangle_property_name_ex(key, &class_name, &prop_name, &prop_len);
+						/* We assume here that a mangled property name is never
+						 * numeric. This is probably a safe assumption, but
+						 * theoretically someone might write an extension with
+						 * private, numeric properties. Well, too bad.
```

Comment:
Nonetheless, one less edge case might be better. Who knows, someone might decide to be annoying.